### PR TITLE
Update to Warzone2100-3.4.0

### PR DESCRIPTION
**Pull request by cozybuild**
Any help is very appreciated :1st_place_medal: 

- [x] Is a migration (config files Warzone-3.3.0) possible?
- [x] Testing: Singleplayer works
- [x] Testing: Multiplayer works

### CODE REVIEW

File: net.wz2100.wz2100.yaml [hunk 1]
```diff
@@ -44,16 +44,33 @@ modules:
...
+    sources:
+      - type: archive
+        sha256: e19fb5e01ea5a707e2a8cb96f537fbd9f3a913d53d804a3265e3aeab3d2064c6
+        url: https://miniupnp.tuxfamily.org/files/miniupnpc-2.1.tar.gz
+
   - name: wz2100
+    buildsystem: "cmake"
+    builddir: true
+    config-opts:
+      - -DENABLE_DOCS=OFF
```
Comment (cozybuild) [inline net.wz2100.wz2100.yaml]: There is a WZ_DISTRIBUTOR variable available; should I set it to `Flatpak`?

Comment (past-due) [reply net.wz2100.wz2100.yaml]: This really helps the Warzone 2100 Project when tracking down bugs. 👍 Probably should be set to `Flathub`.

**Comment by past-due** [comment]
Is there any particular reason why you're building with documentation off? (`-DENABLE_DOCS=OFF`)

Documentation includes the Quickstart guide as well as JS API docs, and should only require adding Asciidoctor to the build process (if it isn't already available).

**Comment by cozybuild** [comment]
I disabled DOCS in my first commit because of the Ruby dependency of Asciidoctor - I have to find out/try how to bring Ruby gems into the build (help is very welcome ;))

**Comment by cozybuild** [comment]
All right, DOCS is enabled now

File: net.wz2100.wz2100.yaml [hunk 2]
```diff
@@ -44,16 +58,33 @@ modules:
...
         path: disable-docs.patch
-      
+
+  - name: libsodium
+    sources:
+      - type: archive
+        url: https://download.libsodium.org/libsodium/releases/libsodium-1.0.18.tar.gz
+        sha256: 6f504490b342a4f8a4c4a02fc9b866cbef8622d5df4e5452b46be121e46636c1
+
+  - name: miniupnpc
```
Comment (past-due) [inline net.wz2100.wz2100.yaml]: It should not be necessary to include this section downloading miniupnpc. WZ's CMake build system will handle this automatically, with an in-tree copy (that is substantially newer than 2.1).

Comment (cozybuild) [reply net.wz2100.wz2100.yaml]: Thank you - I removed it; build is running

**Comment by past-due** [comment]
This looks good to me - is there some way to get this merged? Or, as a [maintainer for the upstream project](https://github.com/orgs/Warzone2100/people), what's the best way to request write permissions to merge this? (We're aiming to move to a more frequent release cadence, so this would be helpful in any event.)

**Comment by da2x** [comment]
> Or, as a maintainer for the upstream project, what's the best way to request write permissions to merge this? (We're aiming to move to a more frequent release cadence, so this would be helpful in any event.)

@past-due As upstream, you can [request ownership of the package](https://github.com/flathub/flathub/wiki/App-Submission#someone-else-has-put-my-app-on-flathubwhat-do-i-do).

**Comment by past-due** [comment]
> * [x]  Is a migration (config files Warzone-3.3.0) possible?

While there are a few quirks migrating 3.3.0 config files to 3.4.0, we are getting them ironed-out for future releases. Is there a standard / recommended way to accomplish this sort of data path migration in the Flathub yaml?

@rndevfx

File: net.wz2100.wz2100.appdata.xml [hunk 3]
```diff
@@ -25,7 +25,7 @@
   <url type="bugtracker">http://developer.wz2100.net/report/1</url>
```
Comment (past-due) [inline net.wz2100.wz2100.appdata.xml]:
```suggestion
  <url type="bugtracker">https://github.com/Warzone2100/warzone2100/issues</url>
```

We've now fully moved bug-tracking to GitHub issues. Updated URL.

File: net.wz2100.wz2100.appdata.xml [hunk 4]
```diff
@@ -25,7 +25,7 @@
...
   <url type="help">http://developer.wz2100.net/wiki/NewFAQ/</url>
```
Comment (past-due) [inline net.wz2100.wz2100.appdata.xml]:
```suggestion
  <url type="help">https://wz2100.net/#faq</url>
```

We're moving away from the old Trac setup - updated URL.

**Review by past-due — CHANGES_REQUESTED** [review]
Couple of small corrections to `appdata.xml`

Also `developer_name` should be:
```diff
- <developer_name>Warzone 2100 team</developer_name>
+ <developer_name>Warzone 2100 Project</developer_name>
```

and `homepage` can be updated to use HTTPS:
```diff
- <url type="homepage">http://wz2100.net/</url>
+ <url type="homepage">https://wz2100.net/</url>
```

Thank you for all of your work on getting this updated! 👍

**Comment by bilelmoussaoui** [comment]
> This looks good to me - is there some way to get this merged? Or, as a [maintainer for the upstream project](https://github.com/orgs/Warzone2100/people), what's the best way to request write permissions to merge this? (We're aiming to move to a more frequent release cadence, so this would be helpful in any event.

Invitation sent. Let me know if someone else from upstream would like write access as well

**Comment by cozybuild** [comment]
> While there are a few quirks migrating 3.3.0 config files to 3.4.0, we are getting them ironed-out for future releases. Is there a standard / recommended way to accomplish this sort of data path migration in the Flathub yaml?

You mean a migration of data in users dir? By default the config files are located in `~/.var/app/net.wz2100.wz2100/data/` and there should be no way to modify the existing config files when upgrading the App, because the App can be installed system-wide or for a single user. We could add a custom shell script as a wrapper for the binary to do the migration of the user files

**Comment by past-due** [comment]
> > While there are a few quirks migrating 3.3.0 config files to 3.4.0, we are getting them ironed-out for future releases. Is there a standard / recommended way to accomplish this sort of data path migration in the Flathub yaml?
> 
> You mean a migration of data in users dir? By default the config files are located in `~/.var/app/net.wz2100.wz2100/data/` and there should be no way to modify the existing config files when upgrading the App, because the App can be installed system-wide or for a single user. We could add a custom shell script as a wrapper for the binary to do the migration of the user files

Ah, I see - sounds like it would work with our built-in config file migration once that's up-and-running then. 👍

**Comment by cozybuild** [comment]
> Couple of small corrections to `appdata.xml`
> 
> Also `developer_name` should be:
> 
> ```diff
> - <developer_name>Warzone 2100 team</developer_name>
> + <developer_name>Warzone 2100 Project</developer_name>
> ```
> 
> and `homepage` can be updated to use HTTPS:
> 
> ```diff
> - <url type="homepage">http://wz2100.net/</url>
> + <url type="homepage">https://wz2100.net/</url>
> ```
> 
> Thank you for all of your work on getting this updated! +1

@past-due I resolved the other requests as well in the commit from yesterday; I cannot resolve this PR because the code is part of your request-text ;)

**Review by past-due — APPROVED** [review]

**Pull request merged**